### PR TITLE
Adding profiling to dask shuffle

### PR DESCRIPTION
**Pull request by arunraman**
Adding benchmark profile for dask shuffle.

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

**Comment by ajschmidt8** [comment]
add to allowlist

**Comment by ajschmidt8** [comment]
add to whitelist

**Comment by jakirkham** [comment]
The CI failures here should be resolved by PR ( https://github.com/rapidsai/dask-cuda/pull/623 ) and PR ( https://github.com/rapidsai/cudf/pull/8342 )

**Comment by arunraman** [comment]
> The CI failures here should be resolved by PR ( #623 ) and PR ( [rapidsai/cudf#8342](https://github.com/rapidsai/cudf/pull/8342) )

Thanks, do I need to re-trigger the test?

**Comment by jakirkham** [comment]
No we can do that. We just have to wait for those PRs to be merged

**Comment by madsbk** [comment]
Thanks @arunraman, good idea including profiling of the explicit-comms execution.

Hope you don't mind, I took the liberty to move the profiling to include both the regular and the explicit-comms execution.

**Comment by quasiben** [comment]
rerun tests

**Comment by jakirkham** [comment]
Still waiting on PR ( https://github.com/rapidsai/dask-cuda/pull/623 ) before CI is fixed

**Comment by pentschev** [comment]
rerun tests

**Review by jakirkham — APPROVED** [review]

**Comment by jakirkham** [comment]
@gpucibot merge

File: dask_cuda/benchmarks/local_cudf_shuffle.py [hunk 1]
```diff
@@ -63,12 +51,20 @@ def run(client, args, n_workers, write_profile=None):
...
-    if args.backend == "dask":
-        took = shuffle_dask(args, df, write_profile)
+    if write_profile is None:
+        ctx = contextlib.nullcontext()
     else:
-        took = shuffle_explicit_comms(args, df)
+        ctx = performance_report(filename=args.profile)
+
+    with ctx:
+        t1 = clock()
```
Comment (jakirkham) [inline dask_cuda/benchmarks/local_cudf_shuffle.py:60]: JFYI we may want to replace this with `perf_counter` (or similar) in the future, but that's unrelated to this change

Comment (jakirkham) [reply dask_cuda/benchmarks/local_cudf_shuffle.py:60]: NVM this is done [above]( https://github.com/rapidsai/dask-cuda/blob/81bbc6f85575826b13b3fb45894b54135514e668/dask_cuda/benchmarks/local_cudf_shuffle.py#L3 )

**Comment by madsbk** [comment]
BTW @arunraman, Dask's performance profile might not include the communication within the explict-comms tasks. @jakirkham do you know what the performance profile capture exactly?

**Comment by madsbk** [comment]
@quasiben, what about [manually created endpoints](https://github.com/rapidsai/dask-cuda/blob/branch-21.06/dask_cuda/explicit_comms/comms.py#L136):
```python 
ep = await distributed.comm.connect(address)
```

**Pull request merged**